hooks/src/index.js

```diff
@@ -138,20 +137,20 @@ export function useReducer(reducer, initialState, init) {
 	/** @type {import('./internal').ReducerHookState} */
 	const hookState = getHookState(currentIndex++, 2);
 	hookState._reducer = reducer;
-	if (!hookState._component) {
+	if (!hookState._internal) {
 		hookState._value = [
 			!init ? invokeOrReturn(undefined, initialState) : init(initialState),
 
 			action => {
 				const nextValue = hookState._reducer(hookState._value[0], action);
 				if (hookState._value[0] !== nextValue) {
 					hookState._value = [nextValue, hookState._value[1]];
-					hookState._component.setState({});
+					hookState._internal._component.setState({});
```

Comment:
A good way forward would be to have an update mechanism on an internal I guess 😅

Reply:
Yea, was thinking we need a way to expose something like `rerenderInternal`. Will come later for sure

Reply:
what about the component._renderer API from Keyed?